src/HomeTask4.java

```diff
@@ -0,0 +1,73 @@
+import java.util.Scanner;
+
+public class HomeTask4 {
+    public static void main(String[] args) {
+        String[] wordsArray;
+        int maxLength, filterNum;
+        String stringFilter;
+        String result = "";
+        String phrase1;
+        boolean flagExit = true;
+        Scanner sc = new Scanner(System.in);
+        System.out.println(" Введите,пожалуйста, 3 слова через запятую : ");
+        phrase1 = sc.nextLine();
+        wordsArray = phrase1.split(",");
+        do{
+            System.out.println(" 1 - maximum length");
+            System.out.println(" 2 - start with");
+            System.out.println(" 3 - end with");
+            System.out.println(" 4 - contains");
+            System.out.println(" 0 - exit");
+            filterNum = sc.nextInt();
+
+            switch (filterNum) {
+                case 1 :
+                    System.out.println("maximum length");
+                    maxLength = sc.nextInt();
+                    for (int i=0;i<wordsArray.length;i++){
+                        if (wordsArray[i].length()<=maxLength){
+                            result += wordsArray[i] + ",";
+                        }
+                    }
+                    break;
+                case 2 :
+                    System.out.println("start with");
+                    stringFilter = sc.next();
+                    for (int i=0;i<wordsArray.length;i++){
+                        if (wordsArray[i].substring(0,stringFilter.length()).equals(stringFilter) ) {
```

Comment:
try to find and use one method for string for checking 'starts with' and 'ends with'